pkg/krmtotf/resource.go

```diff
@@ -322,6 +326,21 @@ func (r *Resource) AllTopLevelFieldsAreImmutableOrComputed() bool {
 	return true
 }
 
+func getObservedStateFromStatus(status map[string]interface{}) map[string]interface{} {
+	observedState, exists, _ := unstructured.NestedMap(status, k8s.ObservedStateFieldName)
+	if !exists {
+		observedState = make(map[string]interface{})
+	}
+	return observedState
+}
+
+func (r *Resource) GetStatusOrObservedState() map[string]interface{} {
+	if k8s.HasComputedFieldsUnderObservedState(r.Kind, r.GroupVersionKind().Version) {
+		return getObservedStateFromStatus(r.Status)
```

Comment:
One way to reduce risk in this PR is to rename r.Status to r.status (just the field, not a refactor).  Then we can be sure that we have looked at every code point that reads r.Status / r.Status.ObservedState

Reply:
I love this trick. Will do it and see if there is anything missing.

Reply:
Verified that it the current PR already covers everything we can identify: #1403.